### ISSUE TITLE
Partial extern members missing warning about DllImport

### ISSUE THREAD

**Issue by jjonescz** [dotnet/roslyn#77029]
**Version Used**: main

**Steps to Reproduce**:

```cs
partial class C
{
    public partial void M1();
    public extern partial void M1();

    public extern void M2();
}
```

**Expected Behavior**: `M1` and `M2` to get the same warnings since they are equivalent.

**Actual Behavior**: Only `M2` gets this warning:

```cs
// /TestProject/Program.cs(6,24): warning CS0626: Method, operator, or accessor 'C.M2()' is marked external and has no attributes on it. Consider adding a DllImport attribute to specify the external implementation.
//     public extern void M2();
Diagnostic(ErrorCode.WRN_ExternMethodNoImplementation, "M2").WithArguments("C.M2()").WithLocation(6, 24)
```

**Notes**: See c2c76c9ef0a01f7fcc7495f0c3696560bd7c5c37 where I started fixing this for partial events and constructors but reverted it since it seems orthogonal to the feature.

**Comment by jaredpar** [comment]
@RikkiGibson is this a scenario that we missed when expanding `partial` members? Or do we have a defined behavior for `extern partial`?

**Comment by RikkiGibson** [comment]
I think we just missed it. I think a regular `extern T M();` which is missing a DllImport is going to do the same thing as an `extern partial T M();` missing the same.

**Comment by jjonescz** [comment]
Yeah, this behavior is for partial methods as well.

Also note that this is a slight breaking change because it adds a warning where it wasn't previously.
